app/src/main/java/com/gophillygo/app/activities/PlacesListActivity.java

```diff
@@ -140,14 +158,48 @@ public boolean onOptionsItemSelected(MenuItem item) {
     }
 
     @NonNull
-    private List<DestinationInfo> getFilteredDestinations() {
-        List<DestinationInfo> filteredDestinations = new ArrayList<>(destinations.size());
-        for (DestinationInfo info : destinations) {
+    private ArrayList<DestinationInfo> getFilteredDestinations() {
+        ArrayList<DestinationInfo> filteredDestinations = new ArrayList<>(destinationInfos.size());
+        for (DestinationInfo info : destinationInfos) {
             if (filter.matches(info)) {
                 filteredDestinations.add(info);
             }
         }
         return filteredDestinations;
     }
 
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        // Update on returning from detail view for just the viewed destination.
+        // Allows keeping current list with scroll position, while refreshing the viewed
+        // destination, in case the user changed its flag.
+        if (requestCode == RETURN_FROM_DETAIL_REQUEST) {
+            Log.d(LOG_LABEL, "Returned to places list from place detail");
+            // update the single list item that may have changed
+            if (detailDestinationId > -1) {
+                final long destinationId = detailDestinationId;
+                detailDestinationId = -1; // reset now it has been handled
+                Log.d(LOG_LABEL, "Update viewed list item with ID " + destinationId);
+                for (int i = 0; i < destinationInfos.size(); i++) {
+                    Destination destination = destinationInfos.get(i).getDestination();
+                    if (destination.getId() == destinationId) {
+                        final int idx = i;
+                        viewModel.getDestination(destinationId).observe(this, (final DestinationInfo destinationInfo) -> {
+                            if (destinationInfo == null || destinationInfo.getDestination() == null) {
+                                Log.w(LOG_LABEL, "No matching destination found for ID " + destinationId);
+                                return;
+                            }
+
+                            Log.d(LOG_LABEL, "Found list index for place " + idx);
+                            destinationInfos.set(idx, destinationInfo);
+                            placesListAdapter.submitList(getFilteredDestinations());
+                        });
+                        break;
+                    }
+                }
+            }
+        } else {
+            Log.w(LOG_LABEL, "Unrecognized requestCode " + requestCode);
+        }
+    }
```

Comment:
As we discussed in person, I think we can remove `onActivityResult()` method now.